### PR TITLE
Change resources requests and limits to TypeMap 

### PR DESCRIPTION
**Pull request by jrhouston**
This PR brings support for extended resources in the `resources` block of the container spec by changing the interface for `limits` and `requests` to be TypeMap.

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/master/CHANGELOG.md):

```release-note
Add support for extended resources
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Review by dak1n1 — APPROVED** [review]
Looks good! The test passes locally and I'm running TC on it here:

https://ci-oss.hashicorp.engineering/viewLog.html?buildId=158241&buildTypeId=Kubernetes_ProviderKubernetesGke118&tab=buildResultsDiv

You can wait for that if you want, or merge it and we can address any other issues in another PR if needed.

Thanks for your work on this!

**Comment by dak1n1** [comment]
From the acceptance tests currently in progress:

```
=== RUN   TestAccKubernetesDeployment_regression
    TestAccKubernetesDeployment_regression: resource_kubernetes_deployment_test.go:913: Step 1/2 error: Error running pre-apply refresh: 
        Error: Unsupported argument
        
          on config442499101/terraform_plugin_test.tf line 68, in resource "kubernetes_deployment" "test":
          68:             requests = {
        
        An argument named "requests" is not expected here. Did you mean to define a
        block of type "requests"?
```

```
=== RUN   TestAccKubernetesDeployment_with_resource_requirements
    TestAccKubernetesDeployment_with_resource_requirements: resource_kubernetes_deployment_test.go:559: Step 1/1 error: Check failed: 4 errors occurred:
        	* Check 3/6 error: kubernetes_deployment.test: Attribute 'spec.0.template.0.spec.0.container.0.resources.0.requests.0.memory' not found
        	* Check 4/6 error: kubernetes_deployment.test: Attribute 'spec.0.template.0.spec.0.container.0.resources.0.requests.0.cpu' not found
        	* Check 5/6 error: kubernetes_deployment.test: Attribute 'spec.0.template.0.spec.0.container.0.resources.0.limits.0.memory' not found
        	* Check 6/6 error: kubernetes_deployment.test: Attribute 'spec.0.template.0.spec.0.container.0.resources.0.limits.0.cpu' not found
        
--- FAIL: TestAccKubernetesDeployment_with_resource_requirements (4.66s)
FAIL
```

**Comment by jrhouston** [comment]
That the regression test is failing here makes sense because this is a breaking change and it's using the previous schema, so it's going to complain that `requests` is not an attribute.

I'll fix that other test though, the paths are just incorrect.

File: kubernetes/resource_kubernetes_pod_test.go [hunk 1]
```diff
@@ -576,10 +576,12 @@ func TestAccKubernetesPod_with_resource_requirements(t *testing.T) {
 				Check: resource.ComposeAggregateTestCheckFunc(
 					testAccCheckKubernetesPodExists("kubernetes_pod.test", &conf),
 					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.container.0.image", imageName),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.container.0.resources.0.requests.0.memory", "50Mi"),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.container.0.resources.0.requests.0.cpu", "250m"),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.container.0.resources.0.limits.0.memory", "512Mi"),
-					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.container.0.resources.0.limits.0.cpu", "500m"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.container.0.resources.0.requests.memory", "50Mi"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.container.0.resources.0.requests.cpu", "250m"),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.container.0.resources.0.requests.ephemeral-storage", "128Mi"),
```
[the remark this answers is not in the document]
Comment (jrhouston) [reply kubernetes/resource_kubernetes_pod_test.go:581]: I'm not sure how you got that error – I've actually added a test now that uses the `nvidia/gpu` and it passes. Perhaps you forgot to put quotes around the key in the map?

```hcl
limits = {
  "nvidia/gpu" = "1"
}
```

**Review by dak1n1 — APPROVED** [review]
Nice! It's all working now. Thanks!

**Pull request merged**